packages/orchestration/src/cosmos-api.ts

```diff
@@ -352,3 +355,49 @@ export type CosmosChainAccountMethods<CCI extends CosmosChainInfo> =
 export type ICQQueryFunction = (
   msgs: JsonSafe<RequestQuery>[],
 ) => Promise<JsonSafe<ResponseQuery>[]>;
+
+/**
+ * Message structure for PFM memo
+ *
+ * @see {@link https://github.com/cosmos/chain-registry/blob/58b603bbe01f70e911e3ad2bdb6b90c4ca665735/_memo_keys/ICS20_memo_keys.json#L38-L60}
+ */
+export interface ForwardInfo {
+  forward: {
+    receiver: ChainAddress['value'];
+    port: IBCPortID;
+    channel: IBCChannelID;
+    /** e.g. '10min' */
+    timeout: string;
+    /** default is 3? */
+    retries: number;
+    next?: {
+      forward: ForwardInfo;
+    };
+  };
+}
+
+/**
+ * Object used to help build MsgTransfer parameters for IBC transfers.
+ *
+ * If `forwardInfo` is present:
+ * - it must be stringified and provided as the `memo` field value for
+ * use with `MsgTransfer`.
+ * - `receiver` will be set to `"pfm"` - purposely invalid bech32. see {@link https://github.com/cosmos/ibc-apps/blob/26f3ad8f58e4ffc7769c6766cb42b954181dc100/middleware/packet-forward-middleware/README.md#minimal-example---chain-forward-a-b-c}
+ */
+export type TransferRoute =
+  | {
+      /** typically, `transfer` */
+      sourcePort: string;
+      sourceChannel: IBCChannelID;
+      token: Coin;
+      receiver: typeof PFM_RECEIVER;
+      /** contains PFM forwarding info */
+      forwardInfo: ForwardInfo;
+    }
+  | {
+      /** typically, `transfer` */
+      sourcePort: string;
+      sourceChannel: IBCChannelID;
+      token: Coin;
+      receiver: string;
+    };
```

Comment:
nice.

could be a bit more concise:

```suggestion
export type TransferRoute = {
      /** typically, `transfer` */
      sourcePort: string;
      sourceChannel: IBCChannelID;
      token: Coin;
    } & (
    | {
      receiver: typeof PFM_RECEIVER;
      /** contains PFM forwarding info */
      forwardInfo: ForwardInfo;
    }
  | {
      receiver: string;
    });
```

Reply:
nice